frontend/src/lib/api/candidate.ts

```diff
@@ -0,0 +1,111 @@
+import {get} from 'svelte/store';
+import {constants} from '$lib/utils/constants';
+import {authContext} from '$lib/utils/authenticationStore';
+
+export const authenticate = async (identifier: string, password: string): Promise<Response> => {
+  const url = new URL(constants.PUBLIC_BACKEND_URL);
+  url.pathname = 'api/auth/local';
+
+  return await fetch(url, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({identifier, password})
+  });
+};
+
+export const register = async (registrationKey: string, password: string): Promise<Response> => {
+  const url = new URL(constants.PUBLIC_BACKEND_URL);
+  url.pathname = 'api/auth/candidate/register';
+
+  return await fetch(url, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({registrationKey, password})
+  });
+};
+
+export const requestForgotPasswordLink = async (email: string): Promise<Response> => {
+  const url = new URL(constants.PUBLIC_BACKEND_URL);
+  url.pathname = 'api/auth/forgot-password';
+
+  return await fetch(url, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({email})
+  });
+};
+
+export const resetPassword = async (code: string, password: string) => {
+  const url = new URL(constants.PUBLIC_BACKEND_URL);
+  url.pathname = 'api/auth/reset-password';
+
+  return await fetch(url, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({
+      code: code,
+      password: password,
+      // We leave it up to the UI to do the validation if the password confirmation is correct
+      passwordConfirmation: password
+    })
+  });
+};
+
+export const checkRegistrationKey = async (registrationKey: string): Promise<Response> => {
+  const url = new URL(constants.PUBLIC_BACKEND_URL);
+  url.pathname = 'api/auth/candidate/check';
+
+  return await fetch(url, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({registrationKey})
+  });
+};
+
+/**
+ * Get the current user's data, including candidate information
+ */
+export const me = async (): Promise<unknown> => {
+  // TODO: Define proper type
+  return request(
+    ['api', 'users', 'me'],
+    new URLSearchParams({
+      'populate[candidate][populate][nominations][populate][party]': 'true',
+      'populate[candidate][populate][nominations][populate][constituency]': 'true',
+      'populate[candidate][populate][party]': 'true'
+    })
+  );
+};
+
+export const request = async <T>(
+  endpoint: string[],
```

Comment:
I like it better this way because I think it is cleaner and easier to provide a dynamic pathname to the function (For example variables in the pathname). However, it doesn't really make a big difference and I suppose it is better to keep the code consistent.